server/service.go

```diff
@@ -357,9 +357,17 @@ func (m *BoostService) handleGetHeader(w http.ResponseWriter, req *http.Request)
 				relays[blockHash] = append(relays[blockHash], relay.String())
 			}
 
-			// Skip if value (fee) is not greater than the current highest value
-			if result.response.Data != nil && responsePayload.Data.Message.Value.Cmp(&result.response.Data.Message.Value) < 1 {
-				return
+			// Compare the bid with already known top bid (if any)
+			if result.response.Data != nil {
```

Comment:
I think it'll be great to add unit tests for these core additions. Perhaps refactor into its function to make it more testable as well. Also happy to help on it after the PR is merged if we want to merge it sooner

Reply:
I'd like that! Please take a stab :)